code/datums/sexcon/sex_actions/masturbate/masturbate_other_anus.dm

```diff
@@ -15,18 +15,29 @@
 	return TRUE
 
 /datum/sex_action/masturbate_other_anus/on_start(mob/living/carbon/human/user, mob/living/carbon/human/target)
-	user.visible_message(span_warning("[user] starts fingering [target]'s butt..."))
+	if(HAS_TRAIT(target, TRAIT_TINY) && !(HAS_TRAIT(user, TRAIT_TINY))) //Fairy on non-fairy will be fucking, otherwise normal
```

Comment:
Make a proc is_fairy() on mob or human and replace all these HAS_TRAIT checks with that. This is frequent enough that making it more readable like that should help readability.

Reply:
There already is a proc for isseelie, the tiny trait check was used in case of future race expansion (not done by me I just mean in general) so they could be given the tiny trait and have this all work for them too. Reusability of code and all that